packages/relay/src/lib/eth.ts

```diff
@@ -644,9 +644,15 @@ export class EthImpl implements Eth {
     if (transaction.gas) {
       transaction.gas = parseInt(transaction.gas);
     }
+
+    if (transaction.data && transaction.input) {
+      throw predefined.INVALID_ARGUMENTS('Cannot accept both input and data fields. Use only one.');
+    }
+
     // Support either data or input. https://ethereum.github.io/execution-apis/api-documentation/ lists input but many EVM tools still use data.
     if (transaction.input && transaction.data === undefined) {
       transaction.data = transaction.input;
+      delete transaction.input;
```

Comment:
correct me if I'm wrong but I think we didn't have a unit test that makes sure the transaction.input's value is transferred to transaction.data, and `transaction.input` is deleted from the original txObject.